libvast/include/vast/system/terminate.hpp

```diff
@@ -31,6 +32,45 @@ struct parallel;
 
 namespace vast::system {
 
+/// Wrapper which extends terminator's lifetime to the response handlers.
+template <class ResponseHandle>
+class termination_result {
+public:
+  termination_result(terminator_actor terminator, ResponseHandle response)
+    : terminator_{std::move(terminator)}, response_{std::move(response)} {
+  }
+
+  template <class ResponseHandler, class ErrorHandler>
+  decltype(auto)
+  then(ResponseHandler responseHandler, ErrorHandler errorHandler) {
+    return response_.then(
+      [f = std::move(responseHandler), t = terminator_](atom::done) mutable {
+        f(atom::done_v);
```

Comment:
```suggestion
        std::move(f)(atom::done_v);
```

Otherwise this will not work with mutable lambdas that capture move-only variables.

Reply:
Can you provide some example or some resource? I have tried to reproduce this with online compiler and it seems to work without the move.

`auto k = std::make_unique<int>(5);   
    auto f = [b = std::move(k)]()mutable{
        b.reset();
    };

    auto f2 = [k= std::move(f)]()mutable{k();};
    f2();`

Reply:
Consider an invocable f that can only be called as an rvalue, e.g.,

```cpp
struct F {
  auto operator() && {}
};

auto g = [f = F{}]() mutable {
  // this works:
  std::move(f)(); 
  // this fails:
  // f();
};
```

The same can occur when your function f tries to mutate its own capture, but then it's implicit rather than explicit:

```c++
auto ptr = std::make_unique<int>();

auto f = [ptr = std::move(ptr)]() mutable {
  ptr.release();
};

auto g = [f = std::move(f)]() mutable {
  // this works:
  std::move(f)(); 
  // this fails:
  // f();
};
```